internal/template/template.go

```diff
@@ -150,6 +150,17 @@ func (mock *{{$mock.MockName}}) {{.Name}}Calls() []struct {
 	return calls
 }
 {{end -}}
+
+{{- if $.IncludeHydrate}}
+// Hydrate replaces any non-mocked methods with the corresponding method from the given implementation.
+func (mock *{{$mock.MockName}}) Hydrate(i {{$mock.InterfaceName}}) {
```

Comment:
I have two thoughts about this:

1. What if the interface, that is being mocked already has a method called `Hydrate`?
2. Maybe we can even find an implementation, that does not need the flag to add hydrate but instead just offer it always. (Backwards compatibility?)

Reply:
> What if the interface, that is being mocked already has a method called Hydrate?

I could add an optional flag to customize the method name defaulting to `Hydrate`? 

> Maybe we can even find an implementation, that does not need the flag to add hydrate but instead just offer it always. (Backwards compatibility?)

We could, but that would mean generated code would change with any upgrade after merging this. That feels somewhat annoying to end users that don't need this. Thoughts?

Reply:
cc @breml; sorry for the long delay getting back to you (holidays).

Any follow-up here?

Reply:
@kpurdon I am not the maintainer of this tool, only a regular user and curious follower of the development. So I leave the decision for the maintainers.